openmc/bounding_box.py

```diff
@@ -99,19 +106,64 @@ def extend(self, padding_distance: float) -> BoundingBox:
 
         Returns
         -------
-        An enlarged bounding box
+        An expanded bounding box
+        """
+        self[0] -= padding_distance
+        self[1] += padding_distance
+        return self
+
+    def expand(self, other_box: BoundingBox) -> BoundingBox:
+        """Expand the box to contain another box
+
+        Parameters
+        ----------
+        other_box : BoundingBox
+            The box used to resize this box
+
+        Returns
+        -------
+        An expanded bounding box
+        """
+        self[0][:] = np.minimum(self.lower_left, other_box.lower_left)
+        self[1][:] = np.maximum(self.upper_right, other_box.upper_right)
+        return self
+
+    def reduce(self, other_box: BoundingBox) -> BoundingBox:
+        """Reduce the box to match the dimensions of the input bounding box
+
+        Parameters
+        ----------
+        other_box : BoundingBox
+            The box used to resize this box
+
+        Returns
+        -------
+        A reduced bounding box
+        """
+        self[0][:] = np.maximum(self.lower_left, other_box.lower_left)
+        self[1][:] = np.minimum(self.upper_right, other_box.upper_right)
+        return self
+
+    @classmethod
+    def inf_inverted_box(cls):
+        """Create an infinite, inverted box. Useful as a starting point for determining
+           geometry bounds for union operations.
+
+        Returns
+        -------
+        An infinitely large bounding box.
+        """
+        infs = np.full((3,), np.inf)
+        return cls(infs, -infs)
 
+    @classmethod
+    def inf_box(cls):
```

Comment:
Personally I think `BoundingBox.infinite()` would be clearer. Also really great idea to add this since evidently it was used quite a bit!